main/release/utils/github.py

```diff
@@ -16,41 +16,57 @@ class GitHub:
 
     def __init__(self):
         self.token = os.environ.get('GITHUB_TOKEN')
-        if os.environ.get('GITHUB_EVENT_NAME') != 'release':
-            raise GitHubException('The action was not triggered on release event')
+
+        ALLOWED_GITHUB_ACTIONS = set(['release', 'workflow_dispatch'])
+        github_event = os.environ.get('GITHUB_EVENT_NAME')
+        if github_event not in ALLOWED_GITHUB_ACTIONS:
+            raise GitHubException(f"The action was not triggered on {ALLOWED_GITHUB_ACTIONS} events (is: '{github_event}')")
+
         with open(os.environ.get('GITHUB_EVENT_PATH')) as file:
             self.event = json.load(file)
 
     def get_release_request(self) -> ReleaseRequest:
         repo = self._get_repository()["full_name"]
         organisation, project = repo.split("/")
-        version = self._get_release()['tag_name']
+
+        # Attempt to get the version number. Depending on the event type,
+        # it can be the release tag or an user provided input.
+        release = self._get_release()
+        version = self.event['inputs']['version'] if release is None else release['tag_name']
+
         # tag shall be like X.X.X.BUILD_NUMBER or X.X.X-MX.BUILD_NUMBER or X.X.X+BUILD_NUMBER (SEMVER)
         version_pattern = re.compile(r'^\d+\.\d+\.\d+(?:-M\d+)?[.+](\d+)$')
         version_match = version_pattern.match(version)
         if version_match is None:
             raise GitHubException('The tag must follow this pattern: X.X.X.BUILD_NUMBER or X.X.X-MX.BUILD_NUMBER or X.X.X+BUILD_NUMBER')
-        branch_name = self._get_release()['target_commitish']
-        if re.compile("^([a-f0-9]{40})$").match(branch_name):
-            branch_name = 'master'
+        DEFAULT_BRANCH = 'master' # TODO Need to fetch the actual branch from Github.
```

Comment:
I will do this in another PR since it's an improvement that is outside the scope of the current one.

Reply:
Nevermind, the change was so trivial that I just did it here.